core/tx_pool.go

```diff
@@ -1721,8 +1722,7 @@ func (pool *TxPool) demoteUnexecutables() {
 //   - Post-Espresso: it ensures balance >= GasFeeCap * gas + value + gatewayFee (2)
 //
 // For non-native tokens(cUSD, cEUR, ...) as feeCurrency:
-//   - Pre-Espresso: it ensures balance > GasPrice * gas + gatewayFee (3)
-//   - Post-Espresso: it ensures balance >= GasFeeCap * gas + gatewayFee (4)
+//   - It executes a static call on debitGasFees, implicitly ensuring balance >= GasFeeCap * gas and that `from` is not on the token's block list
 func ValidateTransactorBalanceCoversTx(tx *types.Transaction, from common.Address, currentState *state.StateDB, currentVMRunner vm.EVMRunner, espresso bool) error {
 	if tx.FeeCurrency() == nil {
 		balance := currentState.GetBalance(from)
```

Comment:
When cleaning up maybe also remove the `if espresso` condition below

Reply:
done